DSharpPlus.Commands/Processors/SlashCommands/SlashCommandProcessor.cs

```diff
@@ -0,0 +1,442 @@
+            if (command.Attributes.OfType<SlashCommandTypesAttribute>().FirstOrDefault() is SlashCommandTypesAttribute slashCommandTypesAttribute && !slashCommandTypesAttribute.ApplicationCommandTypes.Contains(ApplicationCommandType.SlashCommand))
+            {
+                continue;
+            }
+
+            applicationCommands.Add(await this.ToApplicationCommandAsync(command));
+        }
+
+        IReadOnlyList<DiscordApplicationCommand> discordCommands = extension.DebugGuildId is null
+            ? await extension.Client.BulkOverwriteGlobalApplicationCommandsAsync(applicationCommands)
+            : await extension.Client.BulkOverwriteGuildApplicationCommandsAsync(extension.DebugGuildId.Value, applicationCommands);
+
+        Dictionary<ulong, Command> commandsDictionary = [];
+        foreach (DiscordApplicationCommand discordCommand in discordCommands)
+        {
+            if (!extension.Commands.TryGetValue(discordCommand.Name, out Command? command))
+            {
+                foreach (Command aliasedCommand in extension.Commands.Values)
+                {
+                    if (aliasedCommand.Attributes.OfType<DisplayNameAttribute>().FirstOrDefault()?.DisplayName == discordCommand.Name)
+                    {
+                        command = aliasedCommand;
+                        break;
+                    }
+                }
+            }
+
+            if (command is null)
+            {
+                SlashLogging.UnknownCommandName(this._logger, discordCommand.Name, null);
+                continue;
+            }
+
+            commandsDictionary.Add(discordCommand.Id, command);
+        }
+
+        this.Commands = commandsDictionary.ToFrozenDictionary();
+        SlashLogging.RegisteredCommands(this._logger, this.Commands.Count, null);
+    }
+
+    [SuppressMessage("Roslyn", "CA1859", Justification = "Incorrect warning in NET 8-rc2. TODO: Open an issue in dotnet/runtime.")]
+    public async Task<DiscordApplicationCommand> ToApplicationCommandAsync(Command command)
+    {
+        if (this._extension is null)
+        {
+            throw new InvalidOperationException("SlashCommandProcessor has not been configured.");
+        }
+
+        // Translate the command's name and description.
+        IReadOnlyDictionary<string, string> nameLocalizations = new Dictionary<string, string>();
+        IReadOnlyDictionary<string, string> descriptionLocalizations = new Dictionary<string, string>();
+        if (command.Attributes.OfType<SlashLocalizerAttribute>().FirstOrDefault() is SlashLocalizerAttribute localizerAttribute)
+        {
+            AsyncServiceScope scope = this._extension.ServiceProvider.CreateAsyncScope();
+            nameLocalizations = await localizerAttribute.LocalizeAsync(scope.ServiceProvider, $"{command.FullName}.name");
+            descriptionLocalizations = await localizerAttribute.LocalizeAsync(scope.ServiceProvider, $"{command.FullName}.description");
```

Comment:
i really dislike this logic being part of the attribute itself

Reply:
I noticed